File: matlab/solutions/coding_1_approximation/exo12.m
err = [];
for i=1:size(fList,3)
    % 
    fW = perform_wavortho_transf(fList(:,:,i),Jmin,+1, options);
    e = compute_l2err(fW,1);
    err = [err e];
end
sel = 10:round(n*n/10);
Err = err(sel,:);
Err = Err ./ repmat(Err(1,:), [size(Err,1), 1]);
clf;
h = plot(log10(sel), log10(Err));
if using_matlab()  
    set(h, 'LineWidth', lw);
end
title('log_{10}( \epsilon^2[M]/||f||^2 )');
legend('regular', 'phantom', 'lena', 'mandrill');
axis([log10(sel(1)) log10(sel(length(sel))) -7 0]);
if using_matlab()  
    box('on');
end

File: julia/NtSolutions/coding_1_approximation/exo12.jl
figure(figsize = (8, 5))

names = ["regular3", "phantom", "lena", "mandrill"]

for i in 1 : size(fList)[3]
    fW = perform_wavortho_transf(fList[: , : , i], Jmin, + 1, h)
    cR = sort(abs(fW)[:])[end : -1 : 1]
    err = collect(e for e in vecnorm(fList[:, :, i]).^2 - cumsum(cR.^2))
    Err = err[10 : Int(round(n*n/10))]
    plot(log10(10 : Base.div(n*n, 10)), log10(Err/Err[1]), linewidth = 2, label = names[i])
end
    
    
title(L"$\log_{10}(\epsilon^2[M]/ ||f||^2)$")
xlim(1, log10(Base.div(n*n, 10)))
ylim(-7, 0)
legend(loc = 3)
show()
